app/helpers/transformOrderItems.test.js

```diff
@@ -0,0 +1,77 @@
+import { transformOrderItems } from './transformOrderItems';
+
+const additionalService1 = {
+  catalogueItemType: 'Additional Service',
+  itemId: 'C000001-01-A10001-1',
+  provisioningType: 'Declarative',
+};
+
+const additionalService2 = {
+  catalogueItemType: 'Additional Service',
+  itemId: 'C000001-01-A10001-2',
+  provisioningType: 'Declarative',
+};
+
+const associatedService1 = {
+  catalogueItemType: 'Associated Service',
+  itemId: 'C000001-01-08E-1',
+  provisioningType: 'Declarative',
+};
+
+const associatedService2 = {
+  catalogueItemType: 'Associated Service',
+  itemId: 'C000001-01-08E-2',
+  provisioningType: 'Declarative',
+};
+
+const associatedServiceOnDemand = {
+  catalogueItemType: 'Associated Service',
+  itemId: 'C000001-01-08E-3',
+  provisioningType: 'On Demand',
+};
+
+const associatedServicePatient = {
+  catalogueItemType: 'Associated Service',
+  itemId: 'C000001-01-08E-4',
+  provisioningType: 'Patient',
+};
+
+const solution1 = {
+  catalogueItemType: 'Solution',
+  itemId: 'C000001-01-10001-1',
+  provisioningType: 'Declarative',
+};
+
+const solution2 = {
+  catalogueItemType: 'Solution',
+  itemId: 'C000001-01-10001-2',
+  provisioningType: 'Declarative',
+};
+
+const bothSolutions = [solution1, solution2];
+const oneOffAssociatedServices = [associatedService1, associatedService2];
+const recurringAssociatedServices = [associatedServiceOnDemand, associatedServicePatient];
+
+const allItems = [solution1, solution2,
+  associatedService1, associatedService2,
+  associatedServiceOnDemand, associatedServicePatient,
+  additionalService1, additionalService2];
+
+const allRecurringCosts = [solution1, solution2,
+  associatedServiceOnDemand, associatedServicePatient,
+  additionalService1, additionalService2];
+
+describe('transformOrderItems', () => {
+  it.each`
+    orderItems                     | expected
+    ${[associatedService1]}        | ${{ oneOffCosts: [associatedService1], recurringCosts: [] }}
+    ${[additionalService1]}        | ${{ oneOffCosts: [], recurringCosts: [additionalService1] }}
+    ${[solution1]}                 | ${{ oneOffCosts: [], recurringCosts: [solution1] }}
+    ${oneOffAssociatedServices}    | ${{ oneOffCosts: oneOffAssociatedServices, recurringCosts: [] }}
+    ${bothSolutions}               | ${{ oneOffCosts: [], recurringCosts: bothSolutions }}
+    ${recurringAssociatedServices} | ${{ oneOffCosts: [], recurringCosts: recurringAssociatedServices }}
+    ${allItems}                    | ${{ oneOffCosts: oneOffAssociatedServices, recurringCosts: allRecurringCosts }}
```

Comment:
oooh nice I've never come across doing the test like this before. Looks good. However when I run the test I just see this
![Screenshot 2020-06-25 at 15 19 49](https://user-images.githubusercontent.com/9801518/85737484-548e2300-b6f7-11ea-99be-f9f63b4cae03.png)

Perhaps in the test description below we can string interpolate some additional info so it's obvious for example if one of them conditions fails